lang/cs/Org.Apache.REEF.Common/Telemetry/MetricsService.cs

```diff
@@ -65,13 +89,64 @@ public void OnNext(IContextMessage contextMessage)
                     //// For the counters from multiple evaluators with the same counter name, the value should be aggregated here
                     //// We also need to consider failure cases.  
                     _counters[counter.Name] = counter;
+
+                    if (_totalIncrementSinceLastSink.ContainsKey(counter.Name))
+                    {
+                        _totalIncrementSinceLastSink[counter.Name] += counter.Value - c.Value;
+                    }
+                    else
+                    {
+                        _totalIncrementSinceLastSink.Add(counter.Name, counter.Value - c.Value);
+                    }
```

Comment:
I didn't mean to add this to counter. I meant to encapsulate this into some CounterData, so that metricsService class didn't have to worry about it.

---
In reply to: [110988814](https://github.com/apache/reef/pull/1262#discussion_r110988814) [](ancestors = 110988814)